Lib/pdb.py

```diff
@@ -436,6 +438,19 @@ def _cmdloop(self):
             except KeyboardInterrupt:
                 self.message('--KeyboardInterrupt--')
 
+    def _validate_file_mtime(self):
+        """Check if any of the files loaded in pdb have been modified since
+        the last time we saw it. If so, give a warning."""
+        try:
+            filename = self.curframe.f_code.co_filename
+            mtime = os.path.getmtime(filename)
+        except Exception:
+            return
+        if (filename in self._file_mtime_table and
+            mtime != self._file_mtime_table[filename]):
+            self.message(f"*** WARNING: file '{filename}' changed after pdb started")
```

Comment:
I'm not sure this warning is clear about what it means that the file has changed. @pablogsal is our usability guru.

Reply:
I'm open to any suggestions to make this clearer.

Reply:
something like "file was edited, running stale code". Or "file was edited, restart to reload" or something like that. It need to tell the user what this means to them.

Reply:
I changed this to let user know the code is stale. "restart" is not always the right recommendation to users as they might be actively debugging their issue.

Reply:
Is there no way to reload the code from within pdb?

Reply:
reloading code is a very dangerous action. The function being could be deleted from the source file. We are not talking about reloading the code when we "restart the program", it's during debugging. If my code is in function `f`, and in the source file, I already deleted the function, how would that work? I don't think there's a proper solution to deal with it. The best thing is to let the user know that the code executing and the source file they are looking at do not match.

Reply:
restarting the program will run the new code. We can say something like "if you want to execute the new code, you need to restart the program". I don't want encourage them to simply "restart the program" because that might not be what they want to do. They could be debugging a complicated issue and accidentally changed the source file. Restarting the program might lost the current repro state.

Reply:
"running stale code until program restart"?

Reply:
That works too, less wordy.